packages/@atjson/renderer-react/src/index.ts

```diff
@@ -14,30 +14,38 @@ export type AttributesOf<AnnotationClass> = AnnotationClass extends Annotation<
     }
   : never;
 
-export default class ReactRenderer extends Renderer {
-  private componentLookup: {
-    [key: string]: ComponentType<any>;
-  };
+export type ComponentMap = {
+  [key: string]: ComponentType<any>;
+};
+
+const ReactRendererContext = React.createContext<ComponentMap>({});
+
+export const ReactRendererConsumer = ReactRendererContext.Consumer;
 
-  constructor(
-    document: Document,
-    componentLookup: {
-      [key: string]: ComponentType<any>;
+export const ReactRendererProvider = ({
```

Comment:
There is probably a better way to do this, right?

Reply:
If this is a function component, then we can use `React.FC<{ value: ComponentMap }>`

Reply:
Generally, I have found it better to duplicate the type definitions rather than to use a type definition (because VSCode will give a nicer autocomplete)